### PR TITLE
OSDe2e PR Check patch

### PR DESCRIPTION
**Pull request by MrSantamaria**
Changing the PR Check to match CLI execution. 
The eventual goal is to move all our jobs to run from the binary instead of using scripts.

### CODE REVIEW

**Comment by MrSantamaria** [comment]
Oh man that took an embarrassing amount of commits. 

The export GOFLAGS=""
Disabled the hard check of the go vendors from being used in prow. 
More on: https://go.dev/ref/mod#vendoring

To be merged afterwards:
https://github.com/openshift/osde2e/pull/1322

File: ci-operator/config/openshift/osde2e/openshift-osde2e-main.yaml [hunk 1]
```diff
@@ -18,21 +22,34 @@ resources:
...
 
-    ci/prow_pr_check.sh
+    make check
+  container:
+    from: src
+- as: verify
+  commands: |
+    export GOFLAGS=""
+    export REPORT_DIR="$ARTIFACT_DIR"
+    export CONFIGS="aws,prod,e2e-suite"
```
Comment (dustman9000) [inline ci-operator/config/openshift/osde2e/openshift-osde2e-main.yaml:38]: Is there a reason we need to run the full e2e-suite? Is there a config with a smaller set of core tests we can use? This seems a little excessive for a PR check IMO.

Comment (MrSantamaria) [reply ci-operator/config/openshift/osde2e/openshift-osde2e-main.yaml:38]: We don't have anything smaller but we could make one:
The e2e-suite today:
tests:
  testsToRun:
  - '\[Suite\: e2e\]'
  - '\[Suite\: operators\]'
  - '\[Suite\: service-definition\]'
  - '\[Suite\: app-builds\]'

Comment (MrSantamaria) [reply ci-operator/config/openshift/osde2e/openshift-osde2e-main.yaml]: Created a pr-check config in the osde2e repo that only includes the service-definition tests.

**Pull request merged**